q2_metadata/plugin_setup.py

```diff
@@ -59,3 +64,86 @@
                 'visualization supports interactive filtering, sorting, and '
                 'exporting to common file formats.',
 )
+
+ArtificialGrouping = \
+    SemanticType('ArtificialGrouping', variant_of=SampleData.field['type'])
+
+plugin.register_semantic_types(ArtificialGrouping)
+
+
+class ArtificialGroupingFormat(model.TextFileFormat):
```

Comment:
We really need to make qiime2.Metadata a file format that the transformer system can talk to. Then you could just use that.

Reply:
That sounds really handy.